tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -2186,189 +2113,1061 @@ void test_lte_lc_neighbor_cell_measurement_gci_max_length(void)
-	test_gci_cells[9].timing_advance_meas_time = 18446744073709551614U;
-	test_gci_cells[9].measurement_time = 18446744073709551614U;
-	test_gci_cells[9].phys_cell_id = 123;
-	test_gci_cells[9].rsrp = 127;
-	test_gci_cells[9].rsrq = -127;
+	strcpy(at_notif,
+		"%NCELLMEAS: 1,\"00112233\",\"98712\",\"0AB9\",4800,7,63,31,"
+		"456,4800,8,60,29,4,3500,9,99,18,5,5300,11\r\n");
 
-	test_gci_cells[10].mcc = 555;
-	test_gci_cells[10].mnc = 555;
-	test_gci_cells[10].id = 0x0BCDEF01;
-	test_gci_cells[10].tac = 0x0102;
-	test_gci_cells[10].earfcn = 999999;
-	test_gci_cells[10].timing_advance = 65534;
+	lte_lc_callback_count_expected = 1;
+
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT%NCELLMEAS", EXIT_SUCCESS);
+
+	ret = lte_lc_neighbor_cell_measurement(NULL);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	test_event_data[0].type = LTE_LC_EVT_NEIGHBOR_CELL_MEAS;
+	test_event_data[0].cells_info.current_cell.id = LTE_LC_CELL_EUTRAN_ID_INVALID;
+	test_event_data[0].cells_info.ncells_count = 0;
+	test_event_data[0].cells_info.gci_cells_count = 0;
+
+	at_monitor_dispatch(at_notif);
+}
+
+void test_lte_lc_neighbor_cell_measurement_normal_status_incomplete(void)
+{
+	int ret;
+
+	strcpy(at_notif,
+		"%NCELLMEAS: 2,\"00112233\",\"98712\",\"0AB9\",4800,7,63,31,"
+		"456,4800,8,60,29,4,3500,9,99,18,5,5300,11\r\n");
+
+	lte_lc_callback_count_expected = 1;
+
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT%NCELLMEAS", EXIT_SUCCESS);
+
+	ret = lte_lc_neighbor_cell_measurement(NULL);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	test_event_data[0].type = LTE_LC_EVT_NEIGHBOR_CELL_MEAS;
+	test_event_data[0].cells_info.current_cell.mcc = 987;
+	test_event_data[0].cells_info.current_cell.mnc = 12;
+	test_event_data[0].cells_info.current_cell.id = 0x00112233;
+	test_event_data[0].cells_info.current_cell.tac = 0x0AB9;
+	test_event_data[0].cells_info.current_cell.earfcn = 7;
+	test_event_data[0].cells_info.current_cell.timing_advance = 4800;
+	test_event_data[0].cells_info.current_cell.timing_advance_meas_time = 11;
+	test_event_data[0].cells_info.current_cell.measurement_time = 4800;
+	test_event_data[0].cells_info.current_cell.phys_cell_id = 63;
+	test_event_data[0].cells_info.current_cell.rsrp = 31;
+	test_event_data[0].cells_info.current_cell.rsrq = 456;
+	test_event_data[0].cells_info.ncells_count = 2;
+	test_event_data[0].cells_info.gci_cells_count = 0;
+	test_neighbor_cells[0].earfcn = 8;
+	test_neighbor_cells[0].time_diff = 3500;
+	test_neighbor_cells[0].phys_cell_id = 60;
+	test_neighbor_cells[0].rsrp = 29;
+	test_neighbor_cells[0].rsrq = 4;
+	test_neighbor_cells[1].earfcn = 9;
+	test_neighbor_cells[1].time_diff = 5300;
+	test_neighbor_cells[1].phys_cell_id = 99;
+	test_neighbor_cells[1].rsrp = 18;
+	test_neighbor_cells[1].rsrq = 5;
+
+	at_monitor_dispatch(at_notif);
+}
+
+void test_lte_lc_neighbor_cell_measurement_malformed_fail(void)
+{
+	int ret;
+
+	strcpy(at_notif,
+		"%NCELLMEAS:0,\"00112233\",\"98712\",\"0AB9\",4800,7,63,31,456,4800,"
```

Comment:
I have only one nit about the alignment, other than that looks good!

Reply:
These should be all fixed now throughout the file.